src/azul/chalice.py

```diff
@@ -504,34 +497,105 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    @attr.s(auto_attribs=True, frozen=True, kw_only=True)
+    class metric_alarm:  # noqa: N801
+        """
+        Use this decorator to configure a Lambda's metric alarm, specify a
+        custom number of Lambda invocation resulting in a failure or throttling
+        (threshold) and for which period (5 min - 1 day) are necessary to
+        trigger the metric alarm for the decorated event handler function. This
+        decorator cannot be used to decorate view functions, i.e. functions also
+        decorated with ``@app.route``.
+
+        """
+        #: The AWS/Lambdas namespace to be used for this alarm
+        metric: LambdaMetric
+        #: The value used by the alarm, that when excceded triggers the alarm
+        threshold: int = attr.ib(default=0)
+        #: The time in seconds to evalute the alarm threshold
+        period: int = attr.ib(default=300)
+        lambda_name: str | None = attr.ib(default=None)
```

Comment:
Slightly more characters but easier to read IMO
```suggestion
        lambda_name: Optional[str] = attr.ib(default=None)
```

Reply:
And elsewhere